apps/src/lab2/views/dialogs/GenericPrompt.tsx

```diff
@@ -0,0 +1,66 @@
+import React, {useState} from 'react';
+
+import TextField from '@cdo/apps/componentLibrary/textField';
+
+import GenericDialog, {GenericDialogProps} from './GenericDialog';
+
+export type GenericPromptProps = Required<Pick<GenericDialogProps, 'title'>> & {
+  handleConfirm: (prompt: string) => void;
+  handleCancel?: () => void;
+  placeholder?: string;
+};
+
+/**
+ * Generic Prompt dialog used in Lab2 labs.
+ * The title, message, and confirm button text can be customized.
+ * If no confirm button text is provided, the default text is "OK" (translatable).
+ */
+
+type GenericPromptBodyProps = {
+  placeholder?: string;
+  prompt: string;
+  setPrompt: (newPrompt: string) => void;
+};
+
+const GenericPromptBody: React.FunctionComponent<GenericPromptBodyProps> = ({
+  placeholder,
+  prompt,
+  setPrompt,
+}) => {
+  return (
+    <TextField
+      name="prompt-field"
+      placeholder={placeholder}
+      value={prompt}
+      onChange={e => setPrompt(e.target.value)}
+    />
+  );
+};
```

Comment:
Is there a prop I can tack onto this thing to get it to span the container? I haven't figured it out yet.